src/viam/ur/module/ur_arm.cpp

```diff
@@ -1513,13 +1517,45 @@ void URArm::state_::send_noop_() {
 
 void URArm::state_::run_() {
     VIAM_SDK_LOG(info) << "worker thread started";
+
+    // Periodically, collect a limited number of samples of the
+    // duration of our wait latency on the condition variable. We
+    // expect this to fairly well respect the configured
+    // `robot_control_freq_hz`. Report this in the log so that we can
+    // know how well we are doing staying a reliable consumer of arm
+    // data. The logging also serves as a proof of forward progress
+    // for the worker thread.
+    constexpr std::size_t k_num_samples = 100;
+    constexpr auto k_sampling_interval = std::chrono::minutes(5);
+    auto last_sampling_point = std::chrono::steady_clock::now();
```

Comment:
not against per se but not sure how useful it would be since these are snapshots of [0.2;1]s ([500,100]) I would leave it tough just to get an initial sense of perfs. For it to be very useful (like following how the median shifts while sanding does the heavy calcs) we would need to invest a lot more time

Reply:
Yes, agreed, but at least this 1) gives us an ongoing liveness indication, which I think is valuable, and 2) provides some information we didn't have before about how well we are hitting our obligations. We can iterate on it. I would love it if we had a more general program for giving components a way to surface statistics on their behavior!